core/src/banking_stage/transaction_scheduler/prio_graph_scheduler.rs

```diff
@@ -358,12 +358,18 @@ impl PrioGraphScheduler {
     ) {
         let thread_id = self.in_flight_tracker.complete_batch(batch_id);
         for transaction in transactions {
-            let account_locks = transaction.get_account_locks_unchecked();
-            self.account_locks.unlock_accounts(
-                account_locks.writable.into_iter(),
-                account_locks.readonly.into_iter(),
-                thread_id,
-            );
+            let message = transaction.message();
+            let account_keys = message.account_keys();
+            let write_account_locks = account_keys
+                .iter()
+                .enumerate()
+                .filter_map(|(index, key)| message.is_writable(index).then_some(key));
+            let read_account_locks = account_keys
+                .iter()
+                .enumerate()
+                .filter_map(|(index, key)| (!message.is_writable(index)).then_some(key));
```

Comment:
nit: can these useful method be defined on `SanitizedMesssage` or `SanitizedTransaction`?

Reply:
Thought about it, but I'm generally against adding more stuff into `SDK`. Could probably define a function in this file.

In my experimental branch I do have this defined as member fn of the generic tx trait.

Reply:
okay, thanks for explaining. mine isn't strong opinion.